src/player.py

```diff
@@ -106,26 +110,40 @@ def action_drop(self, action):
-                            print(
-                                Fore.YELLOW + "\nIt's not wise to drop your source of light!" + Style.RESET_ALL)
-                            cmd = input('Still want to drop it [y,n]? -> ')
-                        if i.name.lower() != 'lamp' or cmd.lower() == 'y':
-                            self.removeItemFromUser(i)
-                            self.location.addItemToRoom(i)
-                            print(Fore.YELLOW +
-                                  f'\nYou dropped {i}\n' + Style.RESET_ALL)
-                            break
-                        elif cmd.lower() == 'n':
-                            print(Fore.YELLOW +
-                                  f'\nYou kept the {i}\n' + Style.RESET_ALL)
-                            cmd = 'n'
-                            break
-                else:
-                    print(Fore.RED + '\nItem not found\n' + Style.RESET_ALL)
+                if self.location.location.lower() == 'vendor':
+                    for i in self.items:
+                        if commands[1].lower() == i.name.lower():
+                            if hasattr(i, 'value'):
+                                print(
+                                    Fore.YELLOW + f'\nVendor is happy to pay you {i.value} schmekels.\n' + Style.RESET_ALL)
+                                self.schmekels += i.value
+                                self.removeItemFromUser(i)
+                                self.location.addItemToRoom(i)
+                                break
+                    else:
+                        print(Fore.RED + '\nVendor does not want that.\n' +
+                              Style.RESET_ALL)
+                if self.location.location.lower() != 'vendor':
+                    cmd = 'n'
+                    for i in self.items:
+                        if commands[1].lower() == i.name.lower():
+                            if i.name.lower() == 'lamp':
+                                print(
+                                    Fore.YELLOW + "\nIt's not wise to drop your source of light!" + Style.RESET_ALL)
+                                cmd = input('Still want to drop it [y,n]? -> ')
+                            if i.name.lower() != 'lamp' or cmd.lower() == 'y':
+                                self.removeItemFromUser(i)
+                                self.location.addItemToRoom(i)
+                                print(Fore.YELLOW +
+                                      f'\nYou dropped {i}\n' + Style.RESET_ALL)
+                                break
+                            elif cmd.lower() == 'n':
+                                print(Fore.YELLOW +
+                                      f'\nYou kept the {i}\n' + Style.RESET_ALL)
+                                cmd = 'n'
+                                break
+                    else:
+                        print(Fore.RED + '\nItem not found\n' + Style.RESET_ALL)
```

Comment:
That's really cool what you did here, I hope you had fun putting it together. You're demonstrating that you understand the concepts in this assignment.